### PR TITLE
feat(Calendar): support `text` slot

### PR DESCRIPTION
**Pull request by yuyinws**
Sometimes, we want to custom the text render on calendar, like this:

image

This PR add a new `text` slot to implement it.

### CODE REVIEW

File: packages/vant/src/calendar/CalendarDay.tsx [hunk 1]
```diff
@@ -95,11 +95,22 @@ export default defineComponent({
       }
     };
 
+    const renderText = () => {
+      const { item } = props;
+      const { text } = item;
+
+      if (slots['text']) {
+        return <div class={bem('text')}>{slots['text'](props.item)}</div>;
```
Comment (chenjiahan) [inline packages/vant/src/calendar/CalendarDay.tsx:103]: Do we need a `div` wrapper? It seems unnecessary.

File: packages/vant/src/calendar/CalendarDay.tsx [hunk 2]
```diff
@@ -95,11 +95,22 @@ export default defineComponent({
       }
     };
 
+    const renderText = () => {
+      const { item } = props;
+      const { text } = item;
+
+      if (slots['text']) {
```
Comment (chenjiahan) [inline packages/vant/src/calendar/CalendarDay.tsx:102]:
```suggestion
      if (slots.text) {
```

**Comment by mice33** [comment]
I'm really looking forward to this feature getting merged and shipped soon!

File: packages/vant/src/calendar/CalendarDay.tsx [hunk 3]
```diff
@@ -95,11 +95,18 @@ export default defineComponent({
       }
     };
 
+    const renderText = () => {
+      const { item } = props;
+      const { text } = item;
+
+      return <>{slots.text ? slots.text(props.item) : text}</>;
```
Comment (inottn) [inline packages/vant/src/calendar/CalendarDay.tsx:102]:
```suggestion
      return slots.text ? slots.text(props.item) : props.item.text;
```

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
Thank you! Sorry I was a bit busy the last two weeks, I will release a new version today

**Pull request merged**